File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> sequence, int k) {
    vector<int> answer;
    int start=0;
    int f_s=0;
    int f_e=0;
    int total=0;
    int length=10000000;
    for(int i=0;i<sequence.size();i++){
        total+=sequence[i];
        if(total<k)continue;
        if(sequence[i]>k||length==1)break;
        while(total>k){
            total-=sequence[start];
            start++;
        }
        if(total==k){
            if(i-start+1<length){
                length=i-start+1;
                f_s=start;
                f_e=i;
            }
        }
    }
    answer.push_back(f_s);
    answer.push_back(f_e);
    return answer;
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    var ans_array = Infinity
    let answer = [];
    let start = 0;
    let end = 0;
    let result = 0;
    while(end<sequence.length){
        result += sequence[end];
        // console.log(result)
        // console.log(start+" "+end)
        // console.log()
        if(result===k){
            if(ans_array>end-start+1){
                ans_array = end-start+1
                answer = [start,end]
            }
        }
        while(result>k){
            result -= sequence[start++]
            if(result===k){
                if(ans_array>end-start+1){
                    ans_array = end-start+1
                    answer = [start,end]
                }
            }
        }
        end++
    }
    
    return answer;
}